iterator.md

```diff
@@ -0,0 +1,64 @@
+# Iterator Pattern
+
+Iteratoren werden in der C# Entwicklung als Liste oder Arrays verwendet und beispielsweise als Interface `IEnumerable`
+in dem Programmcode hinterlegt werden. Iteratoren, müssen nicht zwangsweise eine zusätzliche Klasse besitzen, sondern können
+ebenfalls direkt aufgerufen werden, siehe unten im Beispiel. Sie geben hier entsprechend mit der `Yield Return`
+Methode nacheinander die Werte zurück, welche definiert werden müssen.Ein Iterator wird in der Regel mit 
+einer `foreach` Anweisung erzeugt und kann nur in einer Methode oder Get Zugriffsmethode vorkommen.
+
+### Direkter Aufruf ohne IEnumerable Klasse
+```csharp
```

Comment:
please take a look at the markdown linter in the README.md There is a whitespace needed between the heading and the codeblock